src/IO/S3Common.cpp

```diff
@@ -133,7 +135,7 @@ AuthSettings AuthSettings::loadFromConfig(const std::string & config_elem, const
 
     return AuthSettings
     {
-        std::move(access_key_id), std::move(secret_access_key),
+        std::move(access_key_id), std::move(secret_access_key), std::move(session_token),
```

Comment:
We should also set `session_token` below in the function `AuthSettings::updateFrom`